### ISSUE TITLE
ty does not respect ABC inheritance

### ISSUE THREAD

**Issue by tonka3000** [astral-sh/ty#704]
### Summary

Currently ty seems not to respect ABC inheritance.

Example:

```py
from abc import ABC
from typing import Any, Dict


class IAppStore(ABC):
    def __init__(self):
        pass

    def write(self, data: Dict[str, Any]): ...
    def read(self) -> Dict[str, Any]: ...
```

ty report `error[invalid-return-type] <filename> Function always implicitly returns `None`, which is not assignable to return type `dict[str, Any]` on the line `def read(self) -> Dict[str, Any]: ...`.

An ABC class can not be instantiated, therefore this should not be marked as issue in my opinion because the API can only be None here. 

### Version

0.0.1-alpha.12 (f7446a6ee 2025-06-25)

**Comment by tonka3000** [comment]
Closed because `@abstractmethod` is the correct solution 😄

**Comment by AlexWaygood** [comment]
I put up https://github.com/astral-sh/ruff/pull/18942 to improve our diagnostics here :-)
